s3-source-connector/src/integration-test/java/io/aiven/kafka/connect/s3/source/IntegrationTest.java

```diff
@@ -200,50 +204,53 @@ void bytesTestBasedOnMaxMessageBytes(final TestInfo testInfo)
         assertThat(records.get(3)).isEqualTo("DD");
         assertThat(records.get(4)).isEqualTo("EE");
+
+        // Verify offset positions
+        verifyOffsetPositions(offsetKeys, 1);
     }
 
     @Test
-    void multiPartUploadBytesTest(final TestInfo testInfo) throws ExecutionException, InterruptedException {
-        final var topicName = IntegrationBase.topicName(testInfo);
+    void multiPartUploadBytesTest() {
         final Map<String, String> connectorConfig = getConfig(basicConnectorConfig(CONNECTOR_NAME), topicName);
 
-        connectRunner.createConnector(connectorConfig);
+        connectRunner.configureConnector(CONNECTOR_NAME, connectorConfig);
         final String partition = "00001";
         final String key = topicName + "-" + partition + "-" + System.currentTimeMillis() + ".txt";
         multipartUpload(TEST_BUCKET_NAME, key);
         // Poll messages from the Kafka topic and verify the consumed data
-        final List<String> records = IntegrationBase.consumeMessages(topicName, 1, KAFKA_CONTAINER);
+        final List<String> records = IntegrationBase.consumeMessages(topicName, 1, connectRunner.getBootstrapServers());
         assertThat(records.get(0)).contains("performanceeeqjz");
     }
 
     @Test
-    void avroTest(final TestInfo testInfo) throws ExecutionException, InterruptedException, IOException {
-        final var topicName = IntegrationBase.topicName(testInfo);
+    void avroTest() throws IOException {
         final Map<String, String> connectorConfig = getConfig(basicConnectorConfig(CONNECTOR_NAME), topicName);
         connectorConfig.put(INPUT_FORMAT_KEY, InputFormat.AVRO.getValue());
-        connectorConfig.put(SCHEMA_REGISTRY_URL, SCHEMA_REGISTRY.getSchemaRegistryUrl());
+        connectorConfig.put(SCHEMA_REGISTRY_URL, connectRunner.getSchemaRegistryServer());
         connectorConfig.put("value.converter", "io.confluent.connect.avro.AvroConverter");
-        connectorConfig.put(VALUE_CONVERTER_SCHEMA_REGISTRY_URL, SCHEMA_REGISTRY.getSchemaRegistryUrl());
+        connectorConfig.put(VALUE_CONVERTER_SCHEMA_REGISTRY_URL, connectRunner.getSchemaRegistryServer());
         connectorConfig.put(VALUE_SERIALIZER, "io.confluent.kafka.serializers.KafkaAvroSerializer");
 
-        connectRunner.createConnector(connectorConfig);
+        connectRunner.configureConnector(CONNECTOR_NAME, connectorConfig);
 
         // Define Avro schema
-        final String schemaJson = "{\n" + "  \"type\": \"record\",\n" + "  \"name\": \"TestRecord\",\n"
-                + "  \"fields\": [\n" + "    {\"name\": \"message\", \"type\": \"string\"},\n"
-                + "    {\"name\": \"id\", \"type\": \"int\"}\n" + "  ]\n" + "}";
+        final String schemaJson = "{\n" + " \"type\": \"record\",\n" + " \"name\": \"TestRecord\",\n"
+                + " \"fields\": [\n" + " {\"name\": \"message\", \"type\": \"string\"},\n"
+                + " {\"name\": \"id\", \"type\": \"int\"}\n" + " ]\n" + "}";
```

Comment:
Haha, we see these clunky, unreadable schema-as-string all the time with Avro!  If you want to improve readability:

```suggestion
        final String schemaJson = ("{'type':'record','name':'TestRecord','fields':[{" +
            "'name':'message','type':'string'},{'name':'id','type':'int'}]}").replace('\'', '"');

```

Reply:
Thanks. Done.